neuro_san/service/http/server/http_sidecar.py

```diff
@@ -181,9 +181,14 @@ def add_agent(self, agent_name: str):
         request_metadata_str: str = " ".join(self.forwarded_request_metadata)
         agent_server_logging: AgentServerLogging = \
             AgentServerLogging(self.server_name_for_logs, request_metadata_str)
-        agent_service: AsyncAgentService = \
-            AsyncAgentService(self.logger, None, agent_name, agent_network_provider, agent_server_logging)
-        self.allowed_agents[agent_name] = agent_service
+        agent_service_provider: AsyncAgentServiceProvider = \
+            AsyncAgentServiceProvider(
+                self.logger,
+                None,
+                agent_name,
+                agent_network_provider,
+                agent_server_logging)
+        self.allowed_agents[agent_name] = agent_service_provider
 
```

Comment:
Instead of AsyncAgentService instances, "allowed_agents" map now contains AsyncAgentServiceProvider instances for lazy instantiation of AsyncAgentService.